### DISCUSSION TITLE
Documents are consistently processed twice from consumption directory

### DISCUSSION THREAD

**Discussion by KingOfDog** [jonaswinkler/paperless-ng#1518] [Support]
My Paperless setup includes a WebDAV directory which is mapped onto the consume directory of Paperless. That way I can easily upload files from my PC to the server.

However, Paperless tries to process the same document multiple times. First, it fails with `File type not supported.`  I assume this is because the document has not yet finished uploading. Though this is not nice, I can live with it.

Then, Paperless starts processing the document for real – only it processes the same document twice. Probably this is because I have two workers for processing. As soon as the first one finishes, the second one can't add the document anymore as it already exists in the database (with the error `[Errno 2] No such file or directory: [path to file...]`). Then Paperless tries to process the already finished document again (also twice) even though it has already been deleted from the consume directory in the previous task and fails with `File not found`.

All of this throttles the speed at which my server could process documents severely. Is this a configuration issue on my side or a bug in Paperless? I couldn't find anything about this problem so far.

---

My setup is based on Docker Compose. For the WebDAV server I use [Dave](https://github.com/micromata/dave).
I haven't changed any options concerning the consumption except for `PAPERLESS_CONSUMER_RECURSIVE`, however I tried out resetting it to `false` and the issue persists.

**Comment by mimek** [comment]
Can you try to set polling of consume directory with environment variable: PAPERLESS_CONSUMER_POLLING=<num> ? I'm having similar behaviour on Synology shared folder, and this setting seems to be helpful. I'm even getting 3 errors regarding one file processing without this polling setting.
